### PR TITLE
feat(logger): config.ini, proxy, network stats

### PR DESCRIPTION
**Pull request by dylanpulver**
This PR enhances debug logging capabilities by adding comprehensive network telemetry data and configuration details. These additions aim to improve diagnostic capabilities without including any private or sensitive information. The key changes include:

**1. Configuration Logging:**
Logs the contents of the config.ini file to help diagnose configuration-related issues.

**2. Proxy Configuration Logging:**
Logs the proxy settings if they were attempted, providing visibility into network configuration.

**3. Network Telemetry Data Collection:**
- Network I/O Statistics: Logs the bytes and packets sent/received.
- Network Speed Test: Measures and logs the download speed by timing a request to a test URL.
- Network Addresses: Logs IP addresses for each network interface.
- Network Connections: Logs active network connections, including local and remote addresses and statuses.
- Network Interface Stats: Logs statistics for each network interface, such as whether it's up, duplex mode, speed, and MTU.

Linear task: https://linear.app/pyup/issue/SFTY-949/always-add-the-configini-content-into-the-debug-output-and-other 

## Summary by CodeRabbit

- **New Features**
	- Introduced network telemetry functionality to collect and display network performance metrics.
	- Added a new dependency on the `psutil` library to enhance system resource interactions.

- **Bug Fixes**
	- Improved logging in debug mode to include configuration and network telemetry data.

- **Tests**
	- Enhanced test coverage for network telemetry functionality with new unit tests to validate data retrieval and error handling. 

- **Chores**
	- Updated testing requirements to include the `psutil` dependency.

### CODE REVIEW

File: safety/cli.py [hunk 1]
```diff
@@ -46,17 +48,89 @@
...
+    # Get network IO statistics
+    net_io = psutil.net_io_counters()
+    network_info['bytes_sent'] = net_io.bytes_sent
+    network_info['bytes_recv'] = net_io.bytes_recv
+    network_info['packets_sent'] = net_io.packets_sent
+    network_info['packets_recv'] = net_io.packets_recv
+
+    # Test network speed (download speed)
+    test_url = "http://example.com"  # A URL to test the download speed
+    start_time = time.time()
```
Comment (jeff-at-safety) [inline safety/cli.py:66]: use `time.perf_counter()` instead of `time.time()` (here and on line 69)

**Review by jeff-at-safety — APPROVED** [review]

**Review by yeisonvargasf — CHANGES_REQUESTED** [review]
Thank you @dylanpulver

Can you implement tests for this change?

File: tests/test_cli.py [hunk 2]
```diff
@@ -563,4 +565,95 @@ def dummy_function():
...
+
+        ctx = Mock()
+        param = Mock()
+        debug = True
+
+        with patch('sys.argv', ['--debug', 'true']):
+            with patch('logging.basicConfig') as mock_basicConfig:
+                with patch('configparser.ConfigParser.items', return_value=[('key', 'value')]), \
+                     patch('configparser.ConfigParser.sections', return_value=['section']):
+                    cli.configure_logger(ctx, param, debug)
```
[the remark this answers is not in the document]
Comment (dylanpulver) [reply tests/test_cli.py]: switched

**Review by yeisonvargasf — APPROVED** [review]
LGTM

**Pull request merged**